kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -6,19 +6,61 @@
 package io.kroxylicious.proxy.config;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import io.kroxylicious.proxy.config.tls.Tls;
 
 import edu.umd.cs.findbugs.annotations.Nullable;
 
+/**
+ * A virtual cluster.
+ *
+ * @param targetCluster the cluster being proxied
+ * @param clusterNetworkAddressConfigProvider virtual cluster network config - deprecated - use a named listener
+ * @param tls deprecated - tls settings for the virtual cluster - deprecated - use a named listener
+ * @param listeners listeners
+ * @param logNetwork if true, network will be logged
+ * @param logFrames if true, kafka rpcs will be logged
+ * @param filters filers.
+ */
 public record VirtualCluster(TargetCluster targetCluster,
-                             @JsonProperty(required = true) ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
-                             @JsonProperty() Optional<Tls> tls,
+                             @Deprecated(forRemoval = true, since = "0.11.0") ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
```

Comment:
If they're using legacy options we should check that this is non-null in the Constructor

Reply:
You fixed that too.